### PR TITLE
Switch from JSXHint to ESLint

### PR DESCRIPTION
**Pull request by alicoding**
JSXHint is deprecated and it was a bit buggy which causes a lot of problem with what we're using here. I'm switching it to ESLint which support both JS and JSX.

### CODE REVIEW

**Comment by alicoding** [comment]
@ScottDowne R?

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+    "rules": {
+        "indent": [
+            2,
+            2
+        ],
+        "quotes": [
+            2,
+            "single"
```
Comment (ScottDowne) [inline .eslintrc]: Hilarious that the file that says we use single quotes is in a file that's all double.

Comment (ScottDowne) [inline .eslintrc]: Same with some other files we have that I guess we don't bother linting, like package.json.

Comment (alicoding) [inline .eslintrc]: What other file? Also we only test .js and .jsx, so any JSON is ignored

**Comment by alicoding** [comment]
it looks like there are many other .jsx file that is not currently linting, but I will try to enable this tomorrow.

File: .eslintrc (hunk 1 above)
Comment (ScottDowne) [inline .eslintrc]: I just find it amusing. Nothing I expect fixed.

**Comment by alicoding** [comment]
@ScottDowne not sure if you said it is good to merge here? Can you confirm if you have time if I can merge this patch?

**Comment by ScottDowne** [comment]
@alicoding I just wanted to run it first. Good to go now.

**Pull request merged**